src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java

```diff
@@ -258,8 +257,35 @@ public Response createDataset(String jsonBody, @PathParam("identifier") String p
             ds.setIdentifier(null);
             ds.setProtocol(null);
             ds.setGlobalIdCreateTime(null);
+            Dataset managedDs = null;
+            try {
+                managedDs = execCommand(new CreateNewDatasetCommand(ds, createDataverseRequest(u)));
+            } catch (WrappedResponse ww) {
```

Comment:
The one thing I wonder about here is if there's a better place for doing this constraint logic so it's all in one place (instead of having to wrap the create call here with a try catch); as I know we have this logic repeated in a few places... Maybe out of scope, but @sekmiller could you at least comment on what you think?

Reply:
I moved the building of the error string to a util so that was able to reduce redundancy quite a bit.